Documentation/API.md

```diff
@@ -696,6 +732,102 @@ Descriptor objects. It is something that could get revisited in the future,
 but will need careful consideration; the bloat/size issues is of the most
 concern because of Swift's common use for mobile applications.
 
+## FieldMask Utilities
+
+### Merging Two Messages
+
+The `merge(from:fieldMask:)` function in Swift Protobuf enables developers 
+to selectively merge fields from one message into another, guided by a `Google_Protobuf_FieldMask`. 
+This method is particularly useful when you need to update only specific 
+fields in a message without affecting others. 
+The `merge` function is available as a method on `Message` types and requires two parameters: 
+the source message (`from`) containing the data to merge 
+and the `fieldMask` that specifies which fields should be updated.
+
+For example, consider a message with the following structure:
+
+```protobuf
+message ExampleMessage {
+  string foo = 1;
+  string bar = 2;
```

Comment:
It would be good to illustrate how FieldMask works when you have nested messages.   I think maybe expand the examples:
```protobuf
message Example {
  string foo = 1;
  string bar = 2;
  Example nested = 3;
}
```

You can then add a short sentence below explaining that `"nested"` in the FieldMask will affect all fields in the nested message, and `"nested.foo"` will only affect the `foo` field of the `nested` message.